### DISCUSSION TITLE
add 'select all' to Right-click menu popup.

### DISCUSSION THREAD

**Discussion by cequalspid** [puddletag/puddletag#868] [Ideas]
I'm unable to click and drag with a mouse to select all.
Just for added convenience when using a mouse for quick tagging and checks.

**Comment by corubba** [comment]
I think drag-selection wasn't ever a thing in puddletag, this isn't a spreadsheet (even if it looks like it). If you want to select everything, there is Ctrl+A; which I personally find the fastest and most intuitive way because that how it works in a lot of places. For partial selection, left-click the first cell, hold Shift, left-click the last cell.

**Comment by cequalspid** [comment]
With all due respect, the program, wonderful and powerful, is a GUI tool. There are times when I can complete certain actions just using a mouse. 

The thing is it somewhat behaves like a spreadsheet, i.e. working on specific columns, and specific number of cells as desired. But one cannot even select whitespace and drag from there to perform a select all function.

I think having at least select-all in the right-click menu would be a nice addition for some workflow. I never use 'play' ctrl-p in the right click menu and wonder how many others use it vs having to ctrl-a by way of keystrokes.

**Reply by corubba** [reply]
The main table can't easily do drag selection. As to why, the short answer is: Because it has drag-and-drop enabled, and that is mutually exclusive with drag selection. But PRs are always welcome.

The right-click menu contains actions to apply to the current selection, which "select all" isn't. If you want mouse-only access to it, there is a "Select All" entry in the "Edit" menu.

**Reply by cequalspid** [reply]
I've used it more and more and this is a powerful program especially when you get the hang of it!
I haven't even begun using plugins/scripts.

**Comment by sandrotosi** [comment]
answered in https://github.com/puddletag/puddletag/discussions/868#discussioncomment-8623326
